OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -2142,7 +2142,7 @@ static void handleNotificationReceived(OSNotificationGenerationJob notifJob, boo
     */
    static void fireNotificationWillShowInForegroundHandlers(OSNotificationGenerationJob notifJob) {
       // No ExtNotificationWillShowInForegroundHandler or AppNotificationWillShowInForegroundHandler was setup, show the notification
-      if (extNotificationWillShowInForegroundHandler == null && appNotificationWillShowInForegroundHandler == null) {
+      if (extNotificationWillShowInForegroundHandler == null && appNotificationWillShowInForegroundHandler == null || !isInForeground()) {
```

Comment:
Would like to see the `isInForeground()` check as its own if statement so this is easier to read.

Reply:
Yeah I like that, agreed